source/simulator/parameters.cc

```diff
@@ -723,6 +723,33 @@ namespace aspect
       prm.declare_entry ("Names of fields", "",
                          Patterns::List(Patterns::Anything()),
                          "A user-defined name for each of the compositional fields requested.");
+      prm.declare_entry ("Advection method of fields", "",
+                         Patterns::Map (Patterns::Anything(),
```

Comment:
why not use a list of length n_compositional_fields with pattern(field|particles)?